### DISCUSSION TITLE
Feature Request: Roles & Permissions

### DISCUSSION THREAD

**Discussion by sagudelo1200** [pocketbase/pocketbase#261] [Q&A]
Hello!
I suggest implementing a more versatile Roles & Permissions system that allows you to create different roles and assign permissions to each one.

I have created a small graphic piece on what I have come up with.

![pocketbase_suggestion](https://user-images.githubusercontent.com/60363274/182498759-fe46f9c2-7b46-485f-b947-93401a1c5eca.png)

**Comment by ganigeorgiev** [comment]
PocketBase already supports roles/groups/teams/etc. type of access controls.
You can add a new field "role" in the user `profles` collection and use it in your API rules. For example:
```
// allows only authenticated users with role="mod"
@request.user.profile.role = 'mod'
```

Since the user profiles collection is just like any other regular collection, you even have the option to allow users from one role to edit the profile data of other users depending on the `profile` API rules.

You could find more information how to add custom profile field and and example with _manager-subordinate_ users relation in https://pocketbase.io/docs/manage-users/#custom-fields

**Reply by sagudelo1200** [reply]
Excellent, the table in the image I leave it as a suggestion for the UI, I think it would be more comfortable to manage them in this way.

TY, You can close the discussion

**Reply by ganigeorgiev** [reply]
While for some use cases a simple role management like in your example can do the job, most of the time you would want more fine grained controls, hence the API rules. For example allowing a specific role to view only the `public` status records:

```
// "mod" users can access any status, while "regular" users can access only those with status "public"
@request.user.profile.role = 'mod' || (@request.user.profile.role = 'regular' && status = 'public')
```

Or allow access to only the owned records:
```
author = @request.user.id
```

---

For now I'm not planning to change the UI since providing multiple ways to manage the same thing could be confusing and hard to maintain.
